kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableReadsSuite.scala

```diff
@@ -463,4 +463,72 @@ class DeltaTableReadsSuite extends AnyFunSuite with TestUtils {
     }
     assert(e.getMessage.contains("Unsupported reader protocol version"))
   }
+
+  //////////////////////////////////////////////////////////////////////////////////
+  // getSnapshotAtVersion end-to-end tests (log segment tests in SnapshotManagerSuite)
+  //////////////////////////////////////////////////////////////////////////////////
+
+  test("getSnapshotAtVersion: basic end-to-end read") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+      (0 to 10).foreach { i =>
+        spark.range(i*10, i*10 + 10).write
+          .format("delta")
+          .mode("append")
+          .save(path)
+      }
+      // Read a checkpoint version
+      checkTable(
+        path = path,
+        expectedAnswer = (0L to 99L).map(TestRow(_)),
+        version = Some(9),
+        expectedVersion = Some(9)
+      )
+      // Read a JSON version
+      checkTable(
+        path = path,
+        expectedAnswer = (0L to 89L).map(TestRow(_)),
+        version = Some(8),
+        expectedVersion = Some(8)
+      )
+      // Read the current version
+      checkTable(
+        path = path,
+        expectedAnswer = (0L to 109L).map(TestRow(_)),
+        version = Some(10),
+        expectedVersion = Some(10)
+      )
+      // Cannot read a version that does not exist
+      val e = intercept[RuntimeException] {
+        Table.forPath(defaultTableClient, path)
+          .getSnapshotAtVersion(defaultTableClient, 11)
+      }
+      assert(e.getMessage.contains(
+        "Trying to load a non-existent version 11. Latest version available is 10"))
+    }
+  }
+
+  test("getSnapshotAtVersion: end-to-end test with truncated delta log") {
+    val path = goldenTablePath("truncated-delta-log")
```

Comment:
i thought we said we'd just use spark?

Reply:
if this isn't clear, we need to write a contribution guide that at least _we_ can follow

Reply:
I thought this was easier since we need to manually truncate the _delta_log and it makes it easy to verify the end-table is what we want

Reply:
I was under the impression that we should just use spark when it's easy and self-defining but it's not a hard rule

Reply:
> I thought this was easier since we need to manually truncate the _delta_log and it makes it easy to verify the end-table is what we want

What about doing that in a separate file is easier? (maybe there's an element/context here I don't understand)

To me, this just makes the test less self describing and less cohesive

Reply:
We can look at the golden table added and see that the expected log files are there/not there. If we do it in the test we are creating and deleting the directory so there's no visibility.

Reply:
Who would be looking at the log files to verify that? We don't do that en-masse in delta-spark

I'm most concerned about the case of: some developer looks at this test and wants to understand it. Having all the code in one spot seems easiest. Having an assert in the code that the file doesn't/does exist seems an easy way to prove it